chain/network/src/time.rs

```diff
@@ -154,3 +194,28 @@ impl Default for FakeClock {
         Self::new(*FAKE_CLOCK_UTC_START)
     }
 }
+
+/// Interval equivalent to tokio::time::Interval with
+/// MissedTickBehavior::Skip.
+pub struct Interval {
+    next: time::Instant,
+    period: time::Duration,
+}
+
+impl Interval {
+    pub fn new(next: time::Instant, period: time::Duration) -> Self {
+        Self { next, period }
+    }
+
+    /// Cancel-safe.
+    pub async fn tick(&mut self, clock: &Clock) {
+        clock.sleep_until(self.next).await;
+        let now = clock.now();
+        self.next = now + self.period
+            - Duration::nanoseconds(
+                ((now - self.next).whole_nanoseconds() % self.period.whole_nanoseconds())
+                    .try_into()
+                    .expect("too much time has elapsed since the interval was supposed to tick"),
```

Comment:
from what I understand - this would happen only if the duration was longer than 100 years (nanoseconds exceeding the u64) - let's mention it in the expect (as otherwise someone reading the code would think that this expect can trigger often :-)

Reply:
true, this is a copy-paste from tokio::time::MissedTickBehavior. I've added a copy of clarification comment as well.